### PR TITLE
Add generate SQW cube

### PR DESCRIPTION
**Pull request by oerc0122**
Add ability to generate a cube of SQW data for simple testing of symmetrise/cut. For even counts these are
`[-X/2 + 0.5 : 1 : X/2 - 0.5]` i.e. for 2 -> `[-0.5 0.5]`
for odd counts these are
`[-floor(X/2):floor(X/2)]` i.e. for 3 -> `[-1 0 1]`
These are then `cut` such that bins contain 1 pixel per bin of evenly spaced bins in a grid

### CODE REVIEW

File: horace_core/sqw/@sqw/generate_cube_sqw.m [hunk 1]
```diff
@@ -0,0 +1,65 @@
+function out = generate_cube_sqw(shape)
+    % Generate a cube of SQW data FOR TESTING CUT/SYMMETRISE
+    % The data will be invalid in most circumstances
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/generate_cube_sqw.m:3]: what does invalid mean in this context?

File: horace_core/sqw/@sqw/generate_cube_sqw.m [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    if isscalar(shape) && isnumeric(shape)
+        ndim = 4;
+        npix = shape^ndim;
+        coords = zeros(9, npix);
+
+        if mod(shape, 2) == 0
+            col = [-shape/2 + 0.5:shape/2 - 0.5];
+        else
+            col = [-floor(shape/2):floor(shape/2)];
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/generate_cube_sqw.m:14]: this is a bit opaque. If shape is restricted to integers, as in the PR main comment, then floor(shape/2) is equivalent to (shape+1)/2 which would be easier to think about for maintenance purposes. It would be useful to comment this at the top of the function and test for it.

if in fact shape is any real value, then again comment at top of function. But is that what you want?

File: horace_core/sqw/@sqw/generate_cube_sqw.m [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        end
+
+        for i = 1:ndim
+            coords(i, :) = repmat(repelem(col, shape^(i-1)), 1, shape^(ndim-i));
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/generate_cube_sqw.m:18]: Be really good to have a description in words of what this does, may be with an example.

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/@sqw/generate_cube_sqw.m (hunk 1 above)
Comment (oerc0122) [reply horace_core/sqw/@sqw/generate_cube_sqw.m:3]: It means that you can't actually use the data for much beyond cuy/symmetrise and will generally result in crashes or nonsense if used outside of those.

File: _test/test_sqw/test_dummy_sqw.m [hunk 4]
```diff
@@ -97,5 +100,33 @@ function test_build_dummy_sqw(obj)
...
             assertElementsAlmostEqual(pix,pix1,'absolute',1.e-7);
         end
+
+        function test_gen_cube_2x2x2x2(obj)
+            tsqw = sqw.generate_cube_sqw(2);
+
+            assertTrue(isa(tsqw,'sqw'));
+            assertEqual(tsqw.pix.num_pixels, 2^4);
+
+            tval = tsqw.pix.coordinates';
```
Comment (cmarooney-stfc) [inline _test/test_sqw/test_dummy_sqw.m:110]: add comment saying test is to ensure no duplicates created
